docs/source/name_inference.rst

```diff
@@ -0,0 +1,428 @@
+.. currentmodule:: torch
+
+.. _name_inference_reference-doc:
+
+Named Tensors operator coverage
+===============================
+
+Please read :ref:`named_tensors-doc` first for an introduction to named tensors.
+
+This document is a reference for *name inference*, a process for how named tensors
+use names to automatically check that APIs are being called correctly at runtime and
+for how operators propagate names from input tensors to outputs.
+
+More formally, name inference consists of the following two steps:
+
+- Check names: an operator may check that certain dimensions names must match.
+- Propagate names: name inference computes and propagates names to output tensors.
+
+Below is a list of all operations that are supported with named tensors
+and their associated name inference rules. If an operation is not here then it
```

Comment:
This sentence isn't really actionable -- I can always try an operation and see what happens, you don't have to tell me that :).

You could say something like "If you don't see an operation listed here, but it would help your use case, please search if an issue has already been filed and if not file one"...